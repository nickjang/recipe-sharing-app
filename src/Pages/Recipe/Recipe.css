.recipe-page {
  text-align: center;
  width: 100%;
}

.recipe-name {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.recipe-name h3 {
  margin-top: 0;
}

.recipe-name h2,
.recipe-name h3 {
  color: var(--color-white);
}

.recipe-page .recipe-ingredients,
.recipe-page .recipe-instructions,
.recipe-page .recipe-information {
  text-align: left;
}

.recipe-page .recipe-information,
.recipe-page .recipe-ingredients,
.recipe-page .recipe-instructions {
  width: 95%;
  word-break: break-word;
  margin: 0rem auto;
}

.recipe-page .recipe-ingredients,
.recipe-page .recipe-instructions {
  margin-top: revert;
}

@media all and (min-width: 580px) {
  .recipe-name h2 {
    margin-left: 7rem;
  }

  .recipe-page .recipe-author {
    text-align: left;
  }

  .recipe-name {
    flex-direction: row;
  }

  .recipe-name h3 {
    margin-top: revert;
  }

  .recipe-page .recipe-author {
    margin: 3rem auto 2rem auto;
  }

  .recipe-page .recipe-author,
  .recipe-page .recipe-information,
  .recipe-page .recipe-ingredients,
  .recipe-page .recipe-instructions {
    width: 70%;
  }
}

@media all and (min-width: 750px) {
  .recipe-page .recipe-author,
  .recipe-page .recipe-information,
  .recipe-page .recipe-ingredients,
  .recipe-page .recipe-instructions {
    width: 50%;
  }
}
