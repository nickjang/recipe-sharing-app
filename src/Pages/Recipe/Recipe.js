import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import RecipeSharingApiService from '../../services/recipe-sharing-api-service';
import './Recipe.css';

class Recipe extends Component {
  handleEdit = (e) => {
    e.preventDefault();

    const recipeId = this.props.match.params.recipeId;
    const { history } = this.props
    history.push(`/recipes/${recipeId}/edit`)
  }

  async componentDidMount() {
    this.setState({ error: '' });
    try {
      const recipeId = this.props.match.params.recipeId;
      const {
        name,
        information,
        ingredients,
        instructions,
        author } = await RecipeSharingApiService.getRecipe(recipeId);
      this.setState({ name, information, ingredients, instructions, author })
    } catch ({ error }) {
      this.setState({ error });
    }
  }

  render() {
    const { name, information, ingredients, instructions, author } = this.state;
    const isAuthor =  != null && author.id === ; ///////////////////////////////////////////////////////
    return (
      <article>
        <div>
          <h2 className='lg-title'>Recipe</h2>
          <span>: </span>
          <h3 className='lg-title'>{name}</h3>
        </div>
        <p>{author.full_name && `By ${author.full_name}`}</p>
        {isAuthor &&
          <button
            className='lg-btn lg-btn-light mt-1 ml-auto'
            type='button'
            onClick={(e) => { this.handleEdit(e) }}
          > Edit recipe
            </button>}
        <p>{information}</p>
        <h3>Ingredients:</h3>
        <ul>
          {ingredients.map((ingredient, index) =>
            <li key={index + '-ingredient'}>{ingredient}</li>
          )}
        </ul>
        <h3>Instructions:</h3>
        <ul>
          {instructions.map((instruction, index) =>
            <li key={index + '-instruction'}>{instruction}</li>
          )}
        </ul>
      </article >
    );
  }
}

export default withRouter(Recipe);