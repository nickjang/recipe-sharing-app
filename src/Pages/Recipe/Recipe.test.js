import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Recipe from './Recipe';

describe(`Recipe component`, () => {
  it('renders Recipe by default', () => {
    const wrapper = shallow(<Recipe />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})