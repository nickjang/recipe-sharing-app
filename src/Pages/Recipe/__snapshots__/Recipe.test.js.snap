// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component renders Recipe by default 1`] = `
<article
  className="recipe-page mb-2"
>
  <Parallax
    bgClassName="react-parallax-bgimage"
    bgImage="https://preppykitchen.com/wp-content/uploads/2019/06/Ultimate-Chocolate-Cake-feature-1200.jpg"
    bgImageAlt="the cat"
    blur={0}
    className=""
    contentClassName=""
    disabled={false}
    strength={200}
  >
    <div
      className="recipe-name"
    >
      <h2
        className="rs-title mr-1"
      >
        Recipe:
      </h2>
      <h3
        className="rs-title"
      />
    </div>
  </Parallax>
  <p
    className="recipe-author note"
  />
  <p
    className="recipe-information rs-body"
  />
  <h3
    className="recipe-ingredients rs-sub-title w-fit mr-auto"
  >
    Ingredients:
  </h3>
  <ul
    className="recipe-ingredients recipe-list note"
  />
  <h3
    className="recipe-instructions rs-sub-title w-fit mr-auto"
  >
    Instructions:
  </h3>
  <ul
    className="recipe-instructions recipe-list note"
  />
</article>
`;
